### ISSUE TITLE
DLA redshift distribution has a peak

### ISSUE THREAD

**Issue by londumas** [igmhub/SaclayMocks#1]
The DLA redshift distribution has a peak at `z~2.3`. This peak is unphysical and not observed in data.
The following plots show:
 - the DLA redshift distribution
 - the host quasar redshift distribution
 - the ratio between the host quasar redshift distribution and all quasar redshift distribution

![DLA_z_distribution](https://user-images.githubusercontent.com/3166436/54290445-96ca3a00-4570-11e9-8a38-f8d591e3a96e.png)

![hostQSODLA_z_distribution](https://user-images.githubusercontent.com/3166436/54290631-eb6db500-4570-11e9-8a86-ab2d46130cfb.png)

![host_quasar_ratio](https://user-images.githubusercontent.com/3166436/54290755-2a036f80-4571-11e9-8fd3-1f0102531d9b.png)

**Comment by londumas** [comment]
Another place where it can be seen is in number vs (z,NHI).

![distribution_NHI](https://user-images.githubusercontent.com/3166436/54291285-f70dab80-4571-11e9-8c8b-39135c0ba74b.png)

**Comment by jmarclegoff** [comment]
The shape of the DLA z distribution seems to be coming from that of the QSO distribution, which is really weird. @TEtourneau this is more in DrawQSO that we should look for the problem.

**Comment by TEtourneau** [comment]
What do you mean the QSO distribution ?
If I plot the QSO z distribution from master.fits file, it seems reasonable:
![image](https://user-images.githubusercontent.com/36950825/54374457-73da7b80-467f-11e9-9706-10c0d575ae82.png)

**Comment by jmarclegoff** [comment]
Ok, so the question is then how from this smooth QSO distribution do we get the QSODLA distribution that Hélion has plotted ?
If z_QSO = 2.2 the proba of a DLA is much higher than if z_QSO=2.3 has clearly displayed in Hélion's third plot ???
May be should involve Javier.

**Comment by TEtourneau** [comment]
I already sent him a mail, but didn't get any answer yet. (I linked him the GitHub url)

**Comment by londumas** [comment]
Here is a code reproducing the error. The effect is seen both in the DLA redshift distribution and in the host quasar distribution, but not in the quasar catalog.

![distrib](https://user-images.githubusercontent.com/3166436/54381900-783d6f00-4654-11e9-9f11-87c43694337f.png)

```
### Get mocks
mock = {}
mock['team'] = 'saclay'
mock['mock_ver'] = 'v4.1'
mock['quick_ver'] = 'quick-0.1'
mock['mock_base'] = '/global/projecta/projectdirs/desi/mocks/lya_forest/'+mock['team']+'/'+mock['mock_ver']+'/'
mock['mock_quick'] = mock['mock_base']+'/'+mock['quick_ver']+'/'

toPlot = {'QSO(master)':mock['mock_base']+'master.fits',
         'DLA(masterDLA)':mock['mock_base']+'/master_DLA.fits',
         'QSODLA(masterDLA)':mock['mock_base']+'/master_DLA.fits',}
redshift_key = {'QSO(master)':'Z_QSO_RSD',
         'DLA(masterDLA)':'Z_DLA_RSD',
         'QSODLA(masterDLA)':'Z_QSO_RSD'}

bins = sp.arange(1.2,4.,0.02)
for name,path in sorted(toPlot.items()):
    h = fitsio.FITS(path)
    z = h[1][redshift_key[name]][:]
    h.close()
    plt.hist(z,bins=bins,histtype='step',label=name,density=True,linewidth=4)
plt.xlabel(r'$z$')
plt.ylabel(r'$\mathrm{\#}$')
plt.legend(fontsize=10)
plt.grid()
```

**Comment by TEtourneau** [comment]
I corrected several "small bugs" such as not integer IDs, all seed fixed to the same value, ...
I rerun the production of DLA and here is the new redshift distribution:
![Screen Shot 2019-03-15 at 2 14 39 PM](https://user-images.githubusercontent.com/36950825/54441327-cd54b000-473c-11e9-94f8-31c9fae94b0f.png)

**Comment by TEtourneau** [comment]
Actually I don't know which correction made the distribution looking good, but it did.

**Comment by londumas** [comment]
fixed in v4.2.

![p3](https://user-images.githubusercontent.com/3166436/54628521-591b5480-4a3b-11e9-938e-f5efbd07433d.png)

![p2](https://user-images.githubusercontent.com/3166436/54628522-591b5480-4a3b-11e9-8ae0-4d41a6e144c0.png)

![p1](https://user-images.githubusercontent.com/3166436/54628523-591b5480-4a3b-11e9-8165-beb19bf4792a.png)
